### PR TITLE
Testing forms

### LINKED ISSUE #43: test forms
there is some cool stuff in #41 already, but there are form properties still to be tested.  we need polymoprhic tests like the one in #41 for:

1. post with arbitrary correct input
1. post with arbitrary incorrect input (fuzzed correct input?)
1. ...?

### PR DESCRIPTION
**Pull request by fisx**
Fixes #43.

### CODE REVIEW

**Comment by fisx** [comment]
test case still fails, but that may be due to a bug in the test or to a bug in the code.  will investigate!

**Comment by fisx** [comment]
Need document `selectCategoryValue` better and merge with master.  Will do that tomorrow.

File: src/Frontend/Page/HomePageWithLoginPrompt.hs [hunk 1]
```diff
@@ -28,10 +28,13 @@ instance ToHtml PageHomeWithLoginPrompt where
     toHtmlRaw = toHtml
     toHtml p = semanticDiv p "PageHomeWithLoginPrompt"
 
+newtype LoginFormData = LoginFormData (ST, ST)
```
Comment (Mikolaj) [inline src/Frontend/Page/HomePageWithLoginPrompt.hs]: Why `newtype` with a pair, and not a curried `data`?

File: src/Frontend/Page/HomePageWithLoginPrompt.hs [hunk 2]
```diff
@@ -28,10 +28,13 @@ instance ToHtml PageHomeWithLoginPrompt where
...
+  deriving (Eq, Ord, Show)
+
 instance FormPageView PageHomeWithLoginPrompt where
-    type FormPageResult PageHomeWithLoginPrompt = (ST, ST)
+    type FormPageResult PageHomeWithLoginPrompt = LoginFormData
 
-    makeForm _ = do
+    makeForm _ = LoginFormData <$> do
```
Comment (Mikolaj) [inline src/Frontend/Page/HomePageWithLoginPrompt.hs]: Is this `do` needed?

File: src/Frontend/Page/HomePageWithLoginPrompt.hs (hunk 1 above)
Comment (fisx) [inline src/Frontend/Page/HomePageWithLoginPrompt.hs]: > > +newtype LoginFormData = LoginFormData (ST, ST)
>
> Why `newtype` with a pair, and not a curried `data`?

you are right, will fix.

**Comment by Mikolaj** [comment]
Really shrewd. Re `pendingWith`, in what sense will the input be invalid? I'm worried types get in the way and prevent us from generating something really outrageous. :) Anyway, LGTM.

**Comment by fisx** [comment]
@Mikolaj: good question.  I will add a comment.

**Pull request merged**